test/extended/util/client.go

```diff
@@ -471,6 +489,12 @@ func (c *CLI) TeardownWorkSpace() {
 		})
 		e2e.Debugf("***%v***", c.workSpacesToDelete)
 		for _, ws := range c.workSpacesToDelete {
+			if len(ws.Namespaces) > 0 {
+				for _, ns := range ws.Namespaces {
+					err := c.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("delete").Args("--server="+ws.ServerURL, "namespace", ns).Execute()
+					e2e.Logf("Deleted workspace/%s's namespace/%s, err: %v", ws.Name, ns, err)
+				}
+			}
```

Comment:
As previous PR's comment, no need to tear down namespaces, because tearing down workspace should automatically tear down namespaces otherwise this should be a bug. If you explicitly tear down namespaces, then even if such a bug happens, our test will escape it. What's more, you do not explicitly tear down the "default" namespace, then why other namespaces need tear down?

If this comment is accepted, also consider `c.workSpacesToDelete[len(c.workSpacesToDelete)-1].Namespaces` removal in .

Reply:
Hmm, from my understanding, we just do our best clean up the resources that created by framwork, if we want to test such scenario I suggest that we could add 1 case check delete the ws and the ns under it should also deleted. The default ns actually is created by the ws created I could enhance it to add it to the deleted list.

Reply:
I think we could keep current logic and see if there's any other change of kcp, delete the ns has no harm or flakes it will make delete the ws much gracer just like we delete one ns with many resources it will take more time and efforts for the backend.

Reply:
For users' gereral usage should delete the ws directly without clean up its ns, I commented it and we could add the clean up ns back if needed later.